src/main/java/com/google/opensesame/user/UserData.java

```diff
@@ -139,4 +139,17 @@ private void loadProjects(ArrayList<String> projectIds) throws IOException {
       projects.add(curProject);
     }
   }
+
+  /**
+   * @param {UserData} user2 the user to compare to
+   * @return an integer 'compatibility score' which represents how many shared interests two users
+   *     have.
+   */
+  public int compatibility(UserData user2) {
+    ArrayList<String> user1Interests = (ArrayList) this.interestTags.clone();
+    ArrayList<String> user2Interests = (ArrayList) user2.interestTags.clone();
+    user1Interests.retainAll(user2Interests);
```

Comment:
nit: [Java streams can be used here](https://www.baeldung.com/java-lists-intersection#strings) to avoid having to clone the lists before calculating the intersection. Not that big of a deal, but I thought it was worth a mention as streams are quite clean!

Reply:
Those look great for this! Thanks for sharing.